src/backend_model.cc

```diff
@@ -167,19 +167,25 @@ TritonModel::Create(
 
   TritonModel* raw_local_model = local_model.get();
 
-  // Model initialization is optional... The TRITONBACKEND_Model
-  // object is this TritonModel object. We must set set shared library
-  // path to point to the backend directory in case the backend
-  // library attempts to load additional shared libaries.
+  // Model initialization is optional... The TRITONBACKEND_Model object is this
+  // TritonModel object.
   if (backend->ModelInitFn() != nullptr) {
+    // We must set set shared library path to point to the backend directory in
+    // case the backend library attempts to load additional shared libaries.
+    // Currently, the set and reset function is effective only on Windows, so
+    // there is no need to set path on non-Windows.
+#ifdef _WIN32
```

Comment:
Why does only the Windows build need to lock and se the library directory now? Also, it might be good to figure out a way to log if ModelInit fails even if resetting the library fails. Not reporting the model initialization failure if the library resetting fails for some reasons seems like undesirable behavior.

Reply:
>Why does only the Windows build need to lock and se the library directory now?

There is no effect for calling `slib->SetLibraryDirectory(backend->Directory())` and `slib->ResetLibraryDirectory()` on non-Windows platform, they are basically no-ops on non-Windows. Notice the `SharedLibrary::Acquire(&slib)` is required before calling "set" and "reset" library, so another thread cannot mess with the library setting while the model initialization is ongoing (on Windows). The mechanism is a global mutex that will remove all concurrency if it is held, so, it is important to not hold this global mutex if we are looking into any speedup with parallel model loading (at least on non-Windows).

> Also, it might be good to figure out a way to log if ModelInit fails even if resetting the library fails. Not reporting the model initialization failure if the library resetting fails for some reasons seems like undesirable behavior.

It is a good point, I think a block scope can be used to set/reset the shared library (on Windows), so the error could be printed immediate as it is returned.

Reply:
Yes.

Reply:
Updated.